pkg/kubelet/cm/memorymanager/policy_static.go

```diff
@@ -992,3 +999,22 @@ func isNUMAAffinitiesEqual(numaAffinity1, numaAffinity2 []int) bool {
 
 	return bitMask1.IsEqual(bitMask2)
 }
+
+func isAffinityViolatingNUMAAllocations(machineState state.NUMANodeMap, mask bitmask.BitMask) bool {
+	maskBits := mask.GetBits()
+	singleNUMAHint := len(maskBits) == 1
+	for _, nodeID := range mask.GetBits() {
+		// the node already used for the memory allocation
+		if !singleNUMAHint && machineState[nodeID].NumberOfAssignments > 0 {
+			// the node used for the single NUMA memory allocation, it cannot be used for the multi NUMA node allocation
+			if len(machineState[nodeID].Cells) == 1 {
+				return true
+			}
+			// the node already used with a different group of nodes, it cannot be used with in the current hint
+			if !areGroupsEqual(machineState[nodeID].Cells, maskBits) {
+				return true
+			}
+		}
+	}
+	return false
+}
```

Comment:
can we align to left and simplify the conditions? something like
```
func isAffinityViolatingNUMAAllocations(machineState state.NUMANodeMap, mask bitmask.BitMask) bool {
	maskBits := mask.GetBits()
	singleNUMAHint := len(maskBits) == 1
	for _, nodeID := range maskBits {
		// the node already used for the memory allocation
		if machineState[nodeID].NumberOfAssignments == 0 {
			continue
		}
		if singleNUMAHint {
			continue
		}
		// the node used for the single NUMA memory allocation, it cannot be used for the multi NUMA node allocation
		if len(machineState[nodeID].Cells) == 1 {
			return true
		}
		// the node already used with a different group of nodes, it cannot be used with in the current hint
		if !areGroupsEqual(machineState[nodeID].Cells, maskBits) {
			return true
		}
	}
	return false
}
```
I believe this function is important enough it should have its own set of tests

Reply:
@ffromani Thank you for your review.
I've addressed your comments PTAL when possible